### DISCUSSION TITLE
PyInstaller can't find rtsp __doc__ file

### DISCUSSION THREAD

**Discussion by ilDeffo** [pyinstaller/pyinstaller#5916] [Help] [answered]
I'm tryng to create an executable file (that I called Executable, for testing purpose) under Ubuntu 20.04, but I get this error when trying to run *./Executable*:

    No such file or directory: '/tmp/_MEIIw0x5s/rtsp/__doc__'

The command I used is 

    pyinstaller main.spec

This is the file main.spec:

    # -*- mode: python ; coding: utf-8 -*-

    block_cipher = None
    
    
    a = Analysis(['myproject/main.py'],
                 pathex=['/home/deffo/Projects/myproject/venv/lib/python3.8/site-packages',
                 '/home/deffo/Projects/myproject/venv/lib/python3.8/site-packages/rtsp'],
                 binaries=[],
                 datas=[],
                 hiddenimports=['rtsp'],
                 hookspath=[],
                 runtime_hooks=[],
                 excludes=[],
                 win_no_prefer_redirects=False,
                 win_private_assemblies=False,
                 cipher=block_cipher,
                 noarchive=False)
    pyz = PYZ(a.pure, a.zipped_data,
                 cipher=block_cipher)
    exe = EXE(pyz,
              a.scripts,
              a.binaries,
              a.zipfiles,
              a.datas,
              [],
              name='Executable',
              debug=False,
              bootloader_ignore_signals=False,
              strip=False,
              upx=True,
              upx_exclude=[],
              runtime_tmpdir=None,
              console=True )
    
  
I don't understand what am I missing

**Comment by bwoodsend** [comment, marked as answer]
Edit your spec putting `from PyInstaller.utils.hooks import collect_data_files` at the top, replace `datas=[]` with `datas=collect_data_files('rtsp')`, then rebuild.

**Reply by bwoodsend** [reply]
PS: You should never add `site-packages` folders to `pathex`. If you've remembered to activate your virtual environment then they will be picked up automatically.

**Reply by ilDeffo** [reply]
It worked, thanks!
